TokenRing/src/main/java/RingProcessor.java

```diff
@@ -0,0 +1,122 @@
+import java.io.File;
+import java.io.IOException;
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.logging.FileHandler;
+import java.util.logging.Logger;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static java.lang.Math.max;
+import static java.lang.Math.min;
+
+/**
+ * In the constructor, the ring is initialized, that is, all the nodes and data on the nodes are created.
+ * In the {@link RingProcessor # startProcessing ()} method, the  ring is started -
+ * the data starts processinf clockwise. Also, logging to {@link RingProcessor#logs}.
+ * happens. All work must be thread safe and handle all possible exceptions. If necessary, you can create your own exception classes.
+ */
+public class RingProcessor {
+    private final int nodesAmount;
+    private final int dataAmount;
+
+    private final File logs;
+
+    private List<Node> nodeList;
+
+    private final Logger logger;
+    FileHandler fileHandler;
+    ArrayList<Thread> threads;

```

Comment:
declarations hall be as a list, further on initialisation you make it as an arraylist. 
With "List<Long> timeList;" below you did right, please correct it here